src/v/cloud_storage/remote.h

```diff
@@ -448,12 +448,14 @@ class remote : public cloud_storage_api {
             if (_promise.has_value()) {
                 _hook.unlink();
                 _promise.reset();
+                _gate_holder.release();
             }
         }
 
     private:
         absl::node_hash_set<const retry_chain_node*> _sources_to_ignore;
         std::unordered_set<api_activity_type> _events_to_ignore;
+        ss::gate::holder _gate_holder;
```

Comment:
Do we need the gate at all? Maybe it should be held by `make_notify_cb` instead rather than on the filter?

Reply:
Or not at all. Since the owner of `make_notify_cb` result usually is already "protected" with a gate.

Anyhow. If this passes all tests I suggest merging and optimizing later.

Reply:
did you find out the answer? im also not sure